packages/container/lib/registry.js

```diff
@@ -723,12 +749,30 @@ Registry.prototype = {
   }
 };
 
+function deprecateResolverFunction(registry) {
+  deprecate('Passing a `resolver` function into a Registry is deprecated. Please pass in a Resolver object with a `resolve` method.',
+            false,
+            { id: 'ember-application.registry-resolver-as-function', until: '3.0.0', url: 'http://emberjs.com/deprecations/v2.x#toc_registry-resolver-as-function' });
+  registry.resolver = {
+    resolve: registry.resolver
+  };
+}
+
 function resolve(registry, normalizedName) {
-  var cached = registry._resolveCache[normalizedName];
+  let cached = registry._resolveCache[normalizedName];
   if (cached) { return cached; }
   if (registry._failCache[normalizedName]) { return; }
 
-  var resolved = registry.resolver(normalizedName) || registry.registrations[normalizedName];
+  let resolved;
+
+  if (registry.resolver) {
+    if (typeof registry.resolver === 'function') {
```

Comment:
Since `registry.resolver` is already taken care of in `Registry`'s constructor, we should be able to remove this extra conditional.  I do not believe that changing the resolver after registry creation is a thing we should support...